docs/core-components-2.md

```diff
@@ -0,0 +1,83 @@
+The AEA framework consists of several core elements, some which are required to run an AEA and others which are optional.
+
+## The advanced elements AEAs use
+
+In <a href="../core-components-1/">Core Components - Part 1</a> we discussed the elements each AEA uses. We will now look at some of the advanced elements each AEA uses.
+
+### Decision Maker
+
+The `DecisionMaker` component manages global agent state updates proposed by the skills and processes the resulting ledger transactions.
+
+It is responsible for the AEA's crypto-economic security and goal management, and it contains the preference and ownership representation of the AEA.
+
+Skills communicate with the decision maker via `InternalMessages`. There exist two types of these: `TransactionMessage` and `StateUpdateMessage`.
+
+The `StateUpdateMessage` is used to initialize the decision maker with preferences and ownership states. It can also be used to update the ownership states in the decision maker if the settlement of transaction takes place off chain.
+
+The `TransactionMessage` is used by a skill to propose a transaction to the decision maker. The performative `TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT` is used by a skill to propose a transaction which the decision maker is supposed to settle on chain. The performative `TransactionMessage.Performative.PROPOSE_FOR_SIGNING` is used by the skill to propose a transaction which the decision maker is supposed to sign and which will be settled later.
```

Comment:
Too many details? Especially the `PROPOSE_FOR_SETTLEMENT` and `PROPOSE_FOR_SIGNING`. Again, depends on what this document wants to be: introductory material, or somewhat specific.

Idea:
> The `TransactionMessage` is used by skill to propose a transaction to the decision-maker. It can be used either for settling the transaction on-chain or to sign a transaction to be used within a negotiation.

Also, the fact that we have two performatives is a bit confusing, it conflicts with the protocol performatives.

Reply:
Yes this needs to be simplified; I forgot